runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -0,0 +1,170 @@
+        String typeUrl = getTypeUrl(msg.getDescriptorForType());
+        // Register the schemas to schema table.
+        if (!classMap.containsKey(typeUrl)) {
+            classMap.put(typeUrl, msg.getClass());
+        }
+    }
+
+    public <K extends Message, V extends Message, M extends Message>
+    Table<K, V, M> openTable(@Nonnull final String namespace,
+                             @Nonnull final String tableName,
+                             @Nonnull final Class<K> kClass,
+                             @Nonnull final Class<V> vClass,
+                             @Nullable final Class<M> mClass,
+                             @Nonnull final TableOptions tableOptions)
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+
+        // Register the schemas to schema table.
+        K defaultKeyMessage = (K) kClass.getMethod("getDefaultInstance").invoke(null);
+        addTypeToClassMap(defaultKeyMessage);
+
+        V defaultValueMessage = (V) vClass.getMethod("getDefaultInstance").invoke(null);
+        addTypeToClassMap(defaultValueMessage);
+
+        M defaultMetadataMessage = null;
+        if (mClass != null) {
+            defaultMetadataMessage = (M) mClass.getMethod("getDefaultInstance").invoke(null);
+            addTypeToClassMap(defaultMetadataMessage);
+        }
+
+        String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
+
+        // Open and return table instance.
+        Table<K, V, M> table = new Table<>(
+                namespace,
+                fullyQualifiedTableName,
+                defaultValueMessage,
+                defaultMetadataMessage,
+                this.runtime,
+                this.protobufSerializer);
+        tableMap.put(fullyQualifiedTableName, (Table<Message, Message, Message>) table);
+        registerTable(namespace,
+                tableName,
+                defaultKeyMessage.getDescriptorForType().toProto(),
+                defaultValueMessage.getDescriptorForType().toProto());
+        return table;
+    }
+
+    public <K extends Message, V extends Message, M extends Message>
+    Table<K, V, M> getTable(String namespace, String tableName) {
+        String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
+        if (!tableMap.containsKey(fullyQualifiedTableName)) {
+            throw new NoSuchElementException(
+                    String.format("No such table found: namespace: %s, tableName: %s", namespace, tableName));
+        }
+        return (Table<K, V, M>) tableMap.get(fullyQualifiedTableName);
+    }
+
+    public void deleteTable(String namespace, String tableName) {
+    }
+
+    public Collection<TableName> listTables(final String namespace) {
+        return registryTable.keySet()
+                .stream()
+                .filter(tableName -> namespace == null || tableName.getNamespace().equals(namespace))
```

Comment:
don't use null

Reply:
namespace comes in via the public api and is stored as such. @Nullable is used to clarify to caller the same.

Reply:
namespace is effectively a constant. Isn't this going to return all-or-nothing?

Reply:
done